main.go

```diff
@@ -256,26 +287,34 @@ func listallusers(ctx context.Context, client *github.Client, org string) ([]*gi
 }
 
 func runTrufflehog(filepath string, reponame string, orgoruser string) error {
-	outputFile1 := "/tmp/results/thog/" + orgoruser + "_" + reponame + "_" + uuid.NewV4().String() + ".txt"
+	outputDir := "/tmp/results/" + orgoruser + "/" + reponame
+	os.MkdirAll(outputDir, 0700)
+	outputFile1 := outputDir + "/" + "truffleHog"
 
 	// open the out file for writing
 	outfile, fileErr := os.OpenFile(outputFile1, os.O_CREATE|os.O_RDWR, 0644)
 	check(fileErr)
 	defer outfile.Close()
 
+	params := []string{"/root/truffleHog/truffleHog/truffleHog.py", filepath, "--rules=/root/truffleHog/rules.json", "--regex"}
+	if *mergeOutput {
+		params = append(params, "--json")
+	}
 	var cmd1 *exec.Cmd
 
 	if *thogEntropy {
-		cmd1 = exec.Command("python", "./truffleHog/truffleHog/truffleHog.py", "--regex", "--entropy=True", filepath)
+		params = append(params, "--entropy=True")
 	} else {
-		cmd1 = exec.Command("python", "./truffleHog/truffleHog/truffleHog.py", "--regex", "--entropy=False", filepath)
+		params = append(params, "--entropy=False")
 	}
+	cmd1 = exec.Command("python", params...)
 
 	// direct stdout to the outfile
 	cmd1.Stdout = outfile
 
 	err1 := cmd1.Run()
-	if err1 != nil {
+	// truffleHog returns an exit code 1 if it finds anything
```

Comment:
This is interesting behavior! From what I remember, this wasn't the case before. Do you know why it does this?

Reply:
I guess they added this behavior so that truffleHog can be included in a CI/CD pipeline. If it finds any secret, it errors out and the whole build fails.